src/CommandingOfficers/AW2/GE/Jess.java

```diff
@@ -0,0 +1,111 @@
+import CommandingOfficers.AW2.AW2Commander;
+import Engine.GameScenario;
+import Engine.Combat.StrikeParams;
+import Engine.UnitMods.UnitDamageModifier;
+import Engine.UnitMods.UnitModifier;
+import Engine.UnitMods.UnitMovementModifier;
+import Engine.UnitMods.UnitTypeFilter;
+import UI.UIUtils;
+import Terrain.MapMaster;
+import Units.Unit;
+import Units.UnitModel;
+
+public class Jess extends AW2Commander
+{
+  private static final long serialVersionUID = 1L;
+
+  private static final CommanderInfo coInfo = new instantiator();
+  public static CommanderInfo getInfo()
+  {
+    return coInfo;
+  }
+  private static class instantiator extends CommanderInfo
+  {
+    private static final long serialVersionUID = 1L;
+    public instantiator()
+    {
+      super("Jess", UIUtils.SourceGames.AW2, UIUtils.GE);
+      infoPages.add(new InfoPage(
+          "Jess (AW2)\n"
+        + "A gallant tank driving CO who fought her way to the top. Often argues with Eagle.\n"
+        + "Vehicular (land) units have superior attack power (+10). Infantry, air and naval units are comparatively weak (-10)."));
+      infoPages.add(new InfoPage(TurboCharge(null, null),
+          "Movement range of (land) vehicles increases by 1 space. Firepower increases (+20), and fuel and ammo supplies are also replenished.\n"
+        + "+10 defense.\n"));
+      infoPages.add(new InfoPage(Overdrive(null, null),
+          "Increase in the attack strength of vehicular (land) units (+40) and 2-space increase in movement range. Also restores fuel and ammo supplies.\n"
+        + "+10 defense.\n"));
+      infoPages.add(new InfoPage(
+            "Hit: Dandelions\n"
+          + "Miss: Unfit COs"));
+      infoPages.add(AW2_MECHANICS_BLURB);
+    }
+    @Override
+    public Commander create(GameScenario.GameRules rules)
+    {
+      return new Jess(rules);
+    }
+  }
+  private static CommanderAbility TurboCharge(Commander commander, CostBasis cb)
+  {
+    return new JessAbility(commander, cb, "Turbo Charge", 3, 1, 20);
+  }
+  private static CommanderAbility Overdrive(Commander commander, CostBasis cb)
+  {
+    return new JessAbility(commander, cb, "Overdrive", 6, 2, 40);
+  }
+
+  public Jess(GameScenario.GameRules rules)
+  {
+    super(coInfo, rules);
+
+    CommanderAbility.CostBasis cb = getGameBasis();
+    addCommanderAbility(TurboCharge(this, cb));
+    addCommanderAbility(Overdrive(this, cb));
+  }
+
+  @Override
+  public void modifyUnitAttack(StrikeParams params)
+  {
+    if( params.attacker.model.isAny(UnitModel.TANK) )
+      params.attackPower += 10;
+    else
+      params.attackPower -= 10;
+  }
+
+
+  private static class JessAbility extends AW2Ability
+  {
+    private static final long serialVersionUID = 1L;
+    UnitTypeFilter moveMod, attMod;
+
+    JessAbility(Commander commander, CostBasis cb, String name, int cost, int tankMove, int tankAttack)
+    {
+      super(commander, name, cost, cb);
+      AIFlags = PHASE_TURN_START;
```

Comment:
So if all her abilities inherit from this, she'll just always use the cheaper one? Not meta, I think.

Reply:
If she has both charged, the AI should SCOP?
Quoth `AIUtils`:
```java
  public static CommanderAbility queueCromulentAbility(Queue<GameAction> q, Army co, int flags)
  {
    CommanderAbility retVal = null;
    ArrayList<CommanderAbility> abilities = co.getReadyAbilities();
    for( CommanderAbility ab : abilities )
    {
      if( flags == (ab.AIFlags & flags) )
      {
        retVal = ab;
      }
    }
    if (null != q && null != retVal)
      q.offer(new GameAction.AbilityAction(retVal));
    return retVal;
  }
```

Reply:
Yeah, if they are in order, which they are. That means she lost quite a bit of materiel in one turn though.

I mean, I'm not sure what the better way is to handle that except maybe to disable the COP? Eh, your call.

Reply:
If any of the DoR COs have a resupply ability, and it does/doesn't work on material, that would give us precedent.
If not... I mean, none of the other resupply abilities currently do, right? The best argument might be that you said her powers are bad, so maybe she needs the help, but I have trouble breaking consistency for that.

Reply:
Greyfield resupplies materials. He's the only other CO in the entire series that does resupply.
But also, Jess powers have buffs to move and attack. Greyfield's *entire* effect (outside global zone) is resupply.
But also², Greyfield has 3 zone, so he can charge really quickly (and 120/150 for copters isn't _nothing_).
> I mean, I'm not sure what the better way is to handle that except maybe to disable the COP? Eh, your call.

The "better way" is to have an AI that knows what a move/attack boost on a limited set of units is good for, and whether hitting the button this turn would be efficient/effective.
But that's wildly not in scope.

The cart AI had some conditions for each CO to decide whether it hit buttons, but that is also not in scope and it's not like I actually know what those are.

Reply:
There are only two resupply abilities in the whole series? I thought there were more, like Andy or someone, I dunno. If there's only one other, and it works for materials, then sure, why not?
Re: Jess having more stuff for her powers than Greyfield - they are from different games with different designs, and I'm not worried about interpretively balancing them.

Reply:
Andy just gives HP. This makes sense, since he'd probably give his dudes fuel if he knew what an airport is.

Fair enough, lol
Thanks.